pkg/test/e2e/e2eutil/zookeepercluster_util.go

```diff
@@ -50,55 +49,53 @@ func CreateCluster(t *testing.T, k8client client.Client, z *api.ZookeeperCluster
+	logger.Info(fmt.Sprintf("created zookeeper cluster: %s\n", zk.Name))
 	return z, nil
 }
 
 // DeleteCluster deletes the ZookeeperCluster CR specified by cluster spec
-func DeleteCluster(t *testing.T, k8client client.Client, z *api.ZookeeperCluster) error {
-	t.Logf("deleting zookeeper cluster: %s", z.Name)
+func DeleteCluster(logger logr.Logger, k8client client.Client, z *api.ZookeeperCluster) error {
+	logger.Info(fmt.Sprintf("deleting zookeeper cluster: %s", z.Name))
 	err := k8client.Delete(goctx.TODO(), z)
 	if err != nil {
 		return fmt.Errorf("failed to delete CR: %v", err)
 	}
-
-	t.Logf("deleted zookeeper cluster: %s", z.Name)
+	logger.Info(fmt.Sprintf("deleted zookeeper cluster: %s", z.Name))
 	return nil
 }
 
 // UpdateCluster updates the ZookeeperCluster CR
-func UpdateCluster(t *testing.T, k8client client.Client, z *api.ZookeeperCluster) error {
-	t.Logf("updating zookeeper cluster: %s", z.Name)
+func UpdateCluster(logger logr.Logger, k8client client.Client, z *api.ZookeeperCluster) error {
+	logger.Info(fmt.Sprintf("updating zookeeper cluster: %s", z.Name))
 	err := k8client.Update(goctx.TODO(), z)
 	if err != nil {
 		return fmt.Errorf("failed to update CR: %v", err)
 	}
-
-	t.Logf("updated zookeeper cluster: %s", z.Name)
+	logger.Info(fmt.Sprintf("updated zookeeper cluster: %s", z.Name))
 	return nil
 }
 
 // GetCluster returns the latest ZookeeperCluster CR
-func GetCluster(t *testing.T, k8client client.Client, z *api.ZookeeperCluster) (*api.ZookeeperCluster, error) {
+func GetCluster(logger logr.Logger, k8client client.Client, z *api.ZookeeperCluster) (*api.ZookeeperCluster, error) {
 	zk := &api.ZookeeperCluster{}
 	err := k8client.Get(goctx.TODO(), types.NamespacedName{Namespace: z.Namespace, Name: z.Name}, zk)
 	if err != nil {
 		return nil, fmt.Errorf("failed to obtain created CR: %v", err)
 	}
-	t.Logf("zk cluster has ready replicas %v", zk.Status.ReadyReplicas)
+	logger.Info(fmt.Sprintf("zk cluster has ready replicas %v", zk.Status.ReadyReplicas))
 	return zk, nil
 }
 
 // WaitForClusterToBecomeReady will wait until all cluster pods are ready
-func WaitForClusterToBecomeReady(t *testing.T, k8client client.Client, z *api.ZookeeperCluster, size int) error {
-	t.Logf("waiting for cluster pods to become ready: %s", z.Name)
+func WaitForClusterToBecomeReady(logger logr.Logger, k8client client.Client, z *api.ZookeeperCluster, size int) error {
+	logger.Info(fmt.Sprintf("waiting for cluster pods to become ready: %s", z.Name))
 	err := wait.Poll(RetryInterval, ReadyTimeout, func() (done bool, err error) {
-		cluster, err := GetCluster(t, k8client, z)
+		cluster, err := GetCluster(logger, k8client, z)
 		if err != nil {
 			return false, err
 		}
 
-		t.Logf("\twaiting for pods to become ready (%d/%d), pods (%v)", cluster.Status.ReadyReplicas, size, cluster.Status.Members.Ready)
+		logger.Info(fmt.Sprintf("\twaiting for pods to become ready (%d/%d), pods (%v)", cluster.Status.ReadyReplicas, size, cluster.Status.Members.Ready))
```

Comment:
could you please remove the `\t` from here

Reply:
Done